src/io/syncsocket.c

```diff
@@ -353,6 +353,39 @@ static void socket_bind(MVMThreadContext *tc, MVMOSHandle *h, MVMString *host, M
     }
 }
 
+MVMint64 socket_get_port(MVMThreadContext *tc, MVMOSHandle *h) {
+    MVMIOSyncSocketData *data = (MVMIOSyncSocketData *)h->body.data;
+
+    /* UV piece, we just need a FD */
+    uv_tcp_t *socket = (uv_tcp_t *) data->handle;
+    uv_os_fd_t fd;
+
+    struct sockaddr_storage name;
+    int error, len = sizeof(struct sockaddr_storage);
+    MVMint64 port = 0;
+
+    error = uv_fileno((uv_handle_t *) socket, &fd);
+
+    if (error != 0)
+        MVM_exception_throw_adhoc(tc, "Failed to get fileno from uv handle: %s", uv_strerror(error));
```

Comment:
After the merge of the `sync-socket-no-uv` then most of the above goes away, because `->handle` becomes (on UNXI-y things) a file descriptor; on Windows it's a `SOCKET`, and the type `Socket` is typedef's to the appropriate thing. So this should be able to be simplified after the merge.